### DISCUSSION TITLE
Xstate model based testing. Do not stop path if it still can continue

### DISCUSSION THREAD

**Discussion by ErlanBazarov** [statelyai/xstate#3354] [Q&A] [answered]
Is there any way to avoid testing paths that still can continue? For example: 
```
reaches state "cart"
  - via ADD_TO_CART
reaches state "ordered"
  - via ADD_TO_CART -> PLACE_ORDER
```
Can we avoid testing "reaches state cart" ?  What is the point of testing it in the first path if we are testing it anyway in the second path?

**Comment by mattpocock** [comment, marked as answer]
I agree! `@xstate/test@alpha` improves on this behaviour. We should be shipping it this week.
